src/sql/workbench/services/accountManagement/browser/accountDialog.ts

```diff
@@ -357,13 +362,20 @@ export class AccountDialog extends Modal {
 			this.vstelemetryService
 		);
 
-		Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).registerViews([{
-			id: newProvider.addedProvider.id,
-			name: newProvider.addedProvider.displayName,
-			ctorDescriptor: new SyncDescriptor(AccountPanel),
-		}], ACCOUNT_VIEW_CONTAINER);
+		// Check if view is registered, if it isn't, register the view
+		if (!Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).getView(newProvider.addedProvider.id)) {
+			try {
+				Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).registerViews([{
+					id: newProvider.addedProvider.id,
+					name: newProvider.addedProvider.displayName,
+					ctorDescriptor: new SyncDescriptor(AccountPanel),
+				}], ACCOUNT_VIEW_CONTAINER);
+				this.registerActions(newProvider.addedProvider.id);
+			} catch (error) {
+				this.logService.debug(error);
```

Comment:
do we expect exceptions here?  Is it OK to swallow the exception with a log message?  i.e. would the entire experience be broken requiring a more prominent error?

Reply:
This shouldn't be needed anymore since you've added the check to get registered providers first.